### PR TITLE
pwx-37485 | feat: add support for nfs backup location in async disaster recovery

### PR DESCRIPTION
**Pull request by strivedi-px**
**What type of PR is this?**
>feature

**What this PR does / why we need it**:
Adds support for NFS location while creating clusterpair for async DR.

**Does this PR change a user-facing CRD or CLI?**:

Yes, adds support creating clusterpair with NFS in storkctl.

**Does this change need to be cherry-picked to a release branch?**:

Yes, `release-24.4.0` branch.

**Pipeline run**
https://jenkins.pwx.dev.purestorage.com/job/Users/job/strivedi/job/storkctl-test/43/console

### CODE REVIEW

**Review by diptiranjanpx — COMMENTED** [review]
Let's add one integration test taking all input parameters for creating clusterpair with NFSlocation.

File: test/integration_test/storkctl_clusterpair_test.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+			}
+		case 1:
+			if err = setSourceKubeConfig(); err != nil {
+				return err
+			}
+		}
+		if err = stork.Instance().DeleteClusterPair(name, namespace); err != nil && !k8s_errors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete cluster pair,error=%s", err)
+		}
+		if err = stork.Instance().DeleteBackupLocation(name, namespace); err != nil && !k8s_errors.IsNotFound(err) {
```
Comment (diptiranjanpx) [inline test/integration_test/storkctl_clusterpair_test.go:270]: DeleteClusterPair will remove it's associated backuplocation and secret, so no need to remove these explicitily.

File: test/integration_test/storkctl_clusterpair_test.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+	}
+
+	// If we are here then the test has passed
+	testResult = testResultPass
+	log.InfoD("Test status at end of %s test: %s", t.Name(), testResult)
+}
+
+// cleanUpClusterPairTestResources cleans up the resources that get created during the cluster pair integration
+// test execution across both the source and destination cluster.
+func cleanUpClusterPairTestResources(name, namespace string) (err error) {
```
Comment (diptiranjanpx) [inline test/integration_test/storkctl_clusterpair_test.go:255]: Let's get rid of these function overall as discussed.

File: pkg/storkctl/clusterpair.go [hunk 3]
```diff
@@ -535,15 +539,35 @@ func newCreateClusterPairCommand(cmdFactory Factory, ioStreams genericclioptions
...
+						util.CheckErr(getMissingParameterError("nfs-server", "Server address missing for NFS"))
+						return
+					}
+					if len(nfsExportPath) == 0 {
+						util.CheckErr(getMissingParameterError("nfs-export-path", "Export Path to be mounted missing for NFS"))
+						return
+					}
+					if nfsTimeoutSeconds < 1 || nfsTimeoutSeconds > 30 {
+						util.CheckErr(fmt.Errorf("--nfs-timeout-seconds valid range is [1 30]"))
+					}
```
Comment (diptiranjanpx) [inline pkg/storkctl/clusterpair.go:557]: Should we return here?

File: pkg/storkctl/clusterpair.go [hunk 4]
```diff
@@ -535,15 +539,35 @@ func newCreateClusterPairCommand(cmdFactory Factory, ioStreams genericclioptions
...
+					if len(nfsServer) == 0 {
+						util.CheckErr(getMissingParameterError("nfs-server", "Server address missing for NFS"))
+						return
+					}
+					if len(nfsExportPath) == 0 {
+						util.CheckErr(getMissingParameterError("nfs-export-path", "Export Path to be mounted missing for NFS"))
+						return
+					}
+					if nfsTimeoutSeconds < 1 || nfsTimeoutSeconds > 30 {
+						util.CheckErr(fmt.Errorf("--nfs-timeout-seconds valid range is [1 30]"))
```
Comment (strivedi-px) [inline pkg/storkctl/clusterpair.go:556]:
```suggestion
						util.CheckErr(fmt.Errorf("--nfs-timeout-seconds valid range is [1 30]"))
						return
```

File: pkg/storkctl/clusterpair.go (hunk 3 above)
Comment (strivedi-px) [reply pkg/storkctl/clusterpair.go:558]: Yes, made the change.

**Review by diptiranjanpx — APPROVED** [review]

**Pull request merged**